week-9/Homework/mandatory/2-exercises/4-music-player.js

```diff
@@ -18,36 +18,113 @@ This means the order the songs are played in will be random, but each song will
 
  */
 
-
 class MusicPlayer {
-    // Add your code here
-
+  // Add your code here
+  constructor() {
+    this.playlist = [];
+    this.playing = false;
+    this.trackIndex = 0;
+  }
+  add(track, artist) {
+    let song = {};
+    song.name = track;
+    song.artist = artist;
+    this.playlist.push(song);
+  }
+
+  play(choice) {
+    if (this.playlist.length == 0) {
+      console.log("You have no songs in your playlist!");
+    } else {
+      if (choice) {
```

Comment:
if you have a choice, it should be validated if it falls between the playlist range, otherwise, it will. fail with an index out of range error.